extensions/notification/NotificationQueuePopulator.js

```diff
@@ -42,6 +42,75 @@ class NotificationQueuePopulator extends QueuePopulatorExtension {
             || key === undefined);
     }
 
+    /**
+     * Get bucket attributes from log entry
+     *
+     * @param {Object} value - log entry object
+     * @return {Object|undefined} - bucket attributes if available
+     */
+    _getBucketAttributes(value) {
+        if (value && value.attributes) {
+            const { error, result } = safeJsonParse(value.attributes);
+            if (error) {
+                return undefined;
+            }
+            return result;
+        }
+        return undefined;
+    }
+
+    /**
+     * Get bucket name from bucket attributes
+     *
+     * @param {Object} value - log entry object
+     * @return {String|undefined} - bucket name if available
+     */
+    _getBucketNameFromAttributes(value) {
+        const attributes = this._getBucketAttributes(value);
+        if (attributes && attributes.name) {
+            return attributes.name;
+        }
+        return undefined;
+    }
+
+    /**
+     * Get notification configuration from bucket attributes
+     *
+     * @param {Object} value - log entry object
+     * @return {Object|undefined} - notification configuration if available
+     */
+    _getBucketNotificationConfiguration(value) {
+        const attributes = this._getBucketAttributes(value);
+        if (attributes && attributes.notificationConfiguration) {
+            return attributes.notificationConfiguration;
+        }
+        return undefined;
+    }
+
+    /**
+     * Process bucket entry from the log
+     *
+     * @param {Object} value - log entry object
+     * @return {undefined}
+     */
+    _processBucketEntry(value) {
+        const bucketName = this._getBucketNameFromAttributes(value);
+        const notificationConfiguration
+            = this._getBucketNotificationConfiguration(value);
+        if (notificationConfiguration &&
+            Object.keys(notificationConfiguration).length > 0) {
+            const bnConfig = {
+                bucket: bucketName,
+                notificationConfiguration,
+            };
+            // bucket notification config is available, update node
+            this.bnConfigManager.setConfig(bucketName, bnConfig);
+            return undefined;
```

Comment:
is it possible to keep the return consistent with the below, either returning the defined below or changing here?